src/features/appBar/MyAppBar.tsx

```diff
@@ -7,12 +7,19 @@ import Button from '@mui/material/Button';
 import DarkThemeToggler from '../darkThemeProvider/DarkThemeToggler';
 import { Location, useLocation } from 'react-router';
 import MyDrawer from './MyDrawer';
-import LoginButton from '../auth/LoginButton';
+import Login from '../auth/Login';
+import MenuBar from '../menuBar/MenuBar';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../store';
 
 type Props = {};
 
 const MyAppBar = (props: Props) => {
     const location = useLocation();
+    const authed = useSelector(
+        (state: RootState) =>
+            state.authReducer.token != null && !state.authReducer.loading
+    );
```

Comment:
From my understanding, this if should be different component (to not pollute MenuAppBar), but whatever